### DISCUSSION TITLE
more complex search in UI for posts?

### DISCUSSION THREAD

**Discussion by itaru2622** [bluesky-social/atproto#2079] [Bluesky Client App]
Is there any plan to support more complex search against posts in UI?

current search in UI seems to support just logical conjunction when multiple query words are given.
but there are more operations in ordinal search, such as disjunction(OR), negation(NOT), and range(from..end for DAYTIME)

moreover, user may want to search posts by focusing specific key, like actor(from)==bsky.app, similar to https://github.com/bluesky-social/social-app/issues/849

**Comment by bnewbold** [comment]
There are tons of unexplored possibilities for search!

What we have now is very minimal, and will probably stay that way until we deliver on our core priorities like composable moderation, federation, and improved auth (eg, OAuth, 2FA).

Our search index, built on opensearch/elasticsearch, is actually quite small and it would not be too difficult for a third-party to index from the firehose and experiment with advanced queries.
